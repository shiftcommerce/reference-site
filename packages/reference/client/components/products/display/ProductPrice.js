// Libraries
import { Component } from 'react'

class ProductPrice extends Component {
  renderPrice (variants) {
    const priceList = variants.map((variant) => (variant.price))
    const minPrice = Number(Math.min(...priceList)).toFixed(2)
    const maxPrice = Number(Math.max(...priceList)).toFixed(2)

    if (minPrice !== maxPrice) {
      return (
        <span>
          &pound;{ minPrice } - &pound;{ maxPrice }
        </span>
      )
    } else {
      return (
        <span>
          &pound;{ maxPrice }
        </span>
      )
    }
  }

  render () {
    return (
      <div className='c-product-price'>
        { this.renderPrice(this.props.variants) }
      </div>
    )
  }
}

export default ProductPrice
