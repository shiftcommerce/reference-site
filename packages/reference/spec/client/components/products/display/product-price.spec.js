// Component
import ProductPrice from '../../../../../client/components/products/display/product-price'

// Fixtures
import ProductWithVariantsWithSimilarPrice from '../../../../fixtures/product-with-variants-with-similar-price'
import ProductWithVariantsWithDifferentPrices from '../../../../fixtures/product-with-variants-with-different-prices'

test('renders correctly if variants have similar price', () => {
  // Act
  const wrapper = mount(
    <ProductPrice product={ProductWithVariantsWithSimilarPrice} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('£10.00')
})

test('renders correctly if variants have different prices', () => {
  // Act
  const wrapper = mount(
    <ProductPrice product={ProductWithVariantsWithDifferentPrices} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('£10.00 - £12.00')
})
