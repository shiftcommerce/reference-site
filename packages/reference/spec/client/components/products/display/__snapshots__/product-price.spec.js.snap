// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if variants have different prices 1`] = `
<ProductPrice
  product={
    Object {
      "max_current_price": "12.00",
      "min_current_price": "10.00",
    }
  }
>
  <div
    className="c-product-price"
  >
    <span>
      £
      10.00
       - £
      12.00
    </span>
  </div>
</ProductPrice>
`;

exports[`renders correctly if variants have similar price 1`] = `
<ProductPrice
  product={
    Object {
      "max_current_price": "10.00",
      "min_current_price": "10.00",
    }
  }
>
  <div
    className="c-product-price"
  >
    <span>
      £
      10.00
    </span>
  </div>
</ProductPrice>
`;
